iceoryx2-cli/iox2-config/src/commands.rs

```diff
@@ -0,0 +1,127 @@
+pub fn print_system_configuration() {
+    const HEADER_COLOR: &str = "\x1b[4;92m";
+    const VALUE_COLOR: &str = "\x1b[0;94m";
+    const DISABLED_VALUE_COLOR: &str = "\x1b[0;90m";
+    const ENTRY_COLOR: &str = "\x1b[0;37m";
+    const DISABLED_ENTRY_COLOR: &str = "\x1b[0;90m";
+    const COLOR_RESET: &str = "\x1b[0m";
+
+    println!("{}posix system configuration{}", HEADER_COLOR, COLOR_RESET);
+    println!();
+    println!(" {}system info{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<SystemInfo>().collect::<Vec<_>>() {
+        println!(
+            "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            i.value(),
+        );
+    }
+
+    println!();
+    println!(" {}limits{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<Limit>().collect::<Vec<_>>() {
+        let limit = i.value();
+        let limit = if limit == 0 {
+            "[ unlimited ]".to_string()
+        } else {
+            limit.to_string()
+        };
+        println!(
+            "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            limit,
+        );
+    }
+
+    println!();
+    println!(" {}options{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<SysOption>().collect::<Vec<_>>() {
+        if i.is_available() {
+            println!(
+                "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        } else {
+            println!(
+                "  {DISABLED_ENTRY_COLOR}{:<50}{COLOR_RESET} {DISABLED_VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        }
+    }
+
+    println!();
+    println!(" {}features{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<Feature>().collect::<Vec<_>>() {
+        if i.is_available() {
+            println!(
+                "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        } else {
+            println!(
+                "  {DISABLED_ENTRY_COLOR}{:<50}{COLOR_RESET} {DISABLED_VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        }
+    }
+
+    println!();
+    println!(" {}process resource limits{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<ProcessResourceLimit>().collect::<Vec<_>>() {
+        println!(
+            "  {ENTRY_COLOR}{:<43}{COLOR_RESET} soft:  {VALUE_COLOR}{:<24}{COLOR_RESET} hard:  {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            i.soft_limit(),
+            i.hard_limit()
+        );
+    }
+}
+
+pub fn show() -> Result<()> {
+    print_system_configuration();
+
+    Ok(())
+}
+
+pub fn generate() -> Result<()> {
+    let default_file_path = Path::new("config/iceoryx2.toml");
```

Comment:
This might cause some issues because it will depend on where the user calls the command. We need a default location for this (https://github.com/eclipse-iceoryx/iceoryx2/issues/442).

@elfenpiff What in the code needs to be adapted if we move the default location to `/etc/iceoryx2/config.toml` or even `.config/iceoryx2/config.toml` ?

Reply:
I think @elBoberido worked on this last and added here the XDG paths?!

Reply:
@elfenpiff No, I just adjusted a misleading warning.

#442 was created just for this, though.

cc @orecham

Reply:
Note to myself: Do not press the comment button too early.

There is the `XDG_CONFIG_DIRS` env variable on Linux which has a `:` separated lists of locations, including `/etc/xdg`. There is also `XDG_CONFIG_DIR`, which should be `$HOME/.config`. See also https://wiki.archlinux.org/title/XDG_Base_Directory.

There is the `dirs` (https://crates.io/crates/dirs) crate with support for multiple OSes. We do not necessarily need to use that crate, but it gives a good overview where to look at which platform.

Reply:
@brosier01 I would only suggest to add an `iceoryx2` subdirectory to the path to keep it separate from the rest of the config files in a user's environment, so something like:

```
$HOME/.config /home/alice/iceoryx2/config.toml
/Users/Alice/Library/Application Support/iceoryx2/config.toml
C:\Users\Alice\AppData\Roaming\iceoryx2\config.toml
```